### ISSUE TITLE
[a11y]: Datepicker Open state & Skeleton violations

### ISSUE THREAD

**Issue by francinelucca** [carbon-design-system/carbon#14224]
### Package

@carbon/react

### Package version

1.33.2

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Accessibility Checker, Latest Deployment

### Description

Playwright testing do not pass for the open and skeleton state currently due to a11y violations:

- Open state
<img width="1165" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40550942/84ce91e7-b8cb-42f8-846c-f34be28734e8">

- Skeleton state
<img width="1189" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40550942/4e168783-9b40-4891-8389-95465f8bf629">

### Reproduction/example

e2e/components/DatePicker/DatePicker-test.avt.e2e.js

### Steps to reproduce

Unskip the skipped tests in e2e/components/DatePicker/DatePicker-test.avt.e2e.js and run playwright test ` npx playwright test e2e/components/DatePicker/DatePicker-test.avt.e2e.js` , notice tests do not pass.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by francinelucca** [comment]
Make sure to unskip tests when this is resolved
